### PR TITLE
Rename `engine` to `backend`

### PR DESCRIPTION
**Pull request by qwerty287**
rename based on https://woodpecker-ci.org/docs/usage/terminiology

### CODE REVIEW

**Comment by xoxys** [comment]
Out of interest, why should we call it `backend` now?

**Comment by 6543** [comment]
It's the "pipeline backend engine"  to short it engine is most apropriate ... and do not colide with the package name :)

**Review by anbraten — APPROVED** [review]
Thx

**Comment by anbraten** [comment]
> It's the "pipeline backend engine" to short it engine is most apropriate ... and do not colide with the package name :)

But what's the `pipeline backend` then? We also call it backend in the docs.

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
https://woodpecker-ci.org/docs/usage/terminiology 

well we do call it backend ...

File: pipeline/backend/types/backend.go [hunk 1]
```diff
@@ -14,6 +14,48 @@
...
+	TailStep(ctx context.Context, step *Step, taskUUID string) (io.ReadCloser, error)
+
+	// DestroyStep destroys the workflow step.
+	DestroyStep(ctx context.Context, step *Step, taskUUID string) error
+
+	// DestroyWorkflow destroys the workflow environment.
+	DestroyWorkflow(ctx context.Context, conf *Config, taskUUID string) error
+}
+
+// BackendInfo represents the reported information of a loaded engine
```
Comment (6543) [inline pipeline/backend/types/backend.go:54]:
```suggestion
// BackendInfo represents the reported information of a loaded backend
```

**Pull request merged**